asset/js/setup.js

```diff
@@ -95,6 +95,21 @@ function setupSiteNav() {
   );
 }
 
+function setupReadingProgress() {
+  const h = document.documentElement;
+  const b = document.body;
+  const st = 'scrollTop';
+  const sh = 'scrollHeight';
+  const progress = document.querySelector('.progress');
+  let scroll;
+
+  document.addEventListener('scroll', () => {
+    scroll = Number(h[st] || b[st]) / Number((h[sh] || b[sh]) - h.clientHeight);
```

Comment:
Is there a reason why we need to use `Number` here?

Reply:
I think it might also be worth splitting this assignment up into several steps. There is a lot of logic going on here that is not immediately obvious.

Reply:
In the previous iteration, I had set `scroll` to the expression which is now separated into 2 lines - defining `scroll` and `scrollPercentage`. This caused some lint issues with the operators. That's why I used `Number`. 
I have now removed the `Number` typecasting.
I have also switched to the `.` notation for accessing the properties.